crowdin/build.gradle

```diff
@@ -106,3 +106,165 @@ task stageLocalizations(type: Zip) {
+            urlConnection.setRequestMethod("POST")
+            urlConnection.setDoInput(true);
+            urlConnection.setDoOutput(true);
+
+            String boundary = 'theseImprovedTranslationStrings_987654321'
+            byte[] boundaryStart = '--theseImprovedTranslationStrings_987654321\r\n'.getBytes("UTF-8")
+            byte[] boundaryEndMultipart = '--theseImprovedTranslationStrings_987654321--'.getBytes("UTF-8")
+            urlConnection.addRequestProperty("Content-Type", "multipart/form-data; charset=UTF-8; boundary=" + boundary);
+
+            /**
+             * Dear Java JEPs
+             *
+             * Why be content with not having an build in API for making rest calls in 2019?
+             */
+            urlConnection.getOutputStream().with { out ->
+
+                out.write(boundaryStart);
+                sendField(out, 'login', CROWDIN_ACCOUNT_USERNAME);
+
+                out.write(boundaryStart);
+                sendField(out, 'account-key', CROWDIN_ACCOUNT_KEY);
+
+                out.write(boundaryStart);
+                sendField(out, 'language', crowdinLanguageCode);
+
+                out.write(boundaryStart);
+                sendField(out, 'import_duplicates', 'true');
+
+                out.write(boundaryStart);
+                sendField(out, 'import_eq_suggestions', 'true');
+
+                out.write(boundaryStart);
+                sendFile(out, "$branch/core/src/com/biglybt/internat/MessagesBundle.properties", stagedFile.newInputStream(), stagedFile.name)
+
+                //ends this multipart
+                out.write(boundaryEndMultipart);
+                out.flush()
+            }
+
+            logger.lifecycle "uploading translation ${stagedFile.name} to Crowdin ..."
+
+            urlConnection.connect();
+
+            logger.quiet urlConnection.getResponseCode() + ' - ' + urlConnection.getResponseMessage()
+            if (urlConnection.getErrorStream()) logger.quiet urlConnection.getErrorStream()?.text
+        }
+
+    }
+
+
+    project.ext.crowdinLanguageCodeFromLocalFileNameConvention = { filename ->
+        def code = [
+                'ar-SA' : '_ar_SA.properties',
+                'bg'    : '_bg_BG.properties',
+                'bn'    : '_bn_BD.properties',
+                'bs'    : '_bs_BA.properties',
+                'ca'    : '_ca_AD.properties',
+                'ceb'   : '_ceb_PH.properties',
+                'cs'    : '_cs_CZ.properties',
+                'da'    : '_da_DK.properties',
+                'de'    : '_de_DE.properties',
+                'el'    : '_el_GR.properties',
+                'en-GB' : '_en_GB.properties',
+                'en-US' : '_en_US_classic.properties', //*
+                'es-ES' : '_es_ES.properties',
+                'es-VE' : '_es_VE.properties',
+                'eu'    : '_eu.properties', //not picked up by Java/BiglyBT... guessing it's Basque ??
```

Comment:
I found it to be three languages that aren't picked up by the ui client. They seem to be in Crowdin, but you're not able to choose it under Settings -> Language